src/browser/service/storage.rs

```diff
@@ -12,33 +12,75 @@ pub type Storage = web_sys::Storage;
 
 #[allow(clippy::module_name_repetitions)]
 pub fn get_storage() -> Option<Storage> {
-    web_sys::window()
-        .expect("get `window`")
-        .local_storage()
-        .ok()
-        .flatten()
+    web_sys::window()?.local_storage().ok().flatten()
 }
 
-/// Create a new store, from a serializable data structure.
-pub fn store_data<T>(storage: &Storage, name: &str, data: &T)
+/// Clear all data in local storage
+pub fn clear(storage: &Storage) {
+    let _ = storage.clear();
+}
+
+/// All of the keys in local storage
+pub fn keys(storage: &Storage) -> Vec<String> {
+    let mut keys = vec![];
+    let length = storage.length().unwrap_or(0);
+    for index in 0..length {
+        if let Ok(Some(key)) = storage.key(index) {
+            keys.push(key);
+        }
+    }
+    keys
+}
+
+/// Things that can go wrong when trying to load data from local storage.
+pub enum LoadError {
+    /// Could not connect to local storage.
+    CouldNotConnect,
+    /// The data could not be decoded from JSON.
+    DecodeError(serde_json::Error),
+    /// There is no data for that key.
+    NoData,
+}
+
+/// Load a JSON deserializable data structure from local storage.
+pub fn load_data<T>(storage: &Storage, key: &str) -> Result<T, LoadError>
 where
-    T: serde::Serialize,
+    T: serde::de::DeserializeOwned,
 {
-    let serialized = serde_json::to_string(&data).expect("serialize for `LocalStorage`");
-    storage
-        .set_item(name, &serialized)
-        .expect("save into `LocalStorage`");
+    let item = storage
+        .get_item(key)
+        .map_err(|_| LoadError::CouldNotConnect)?;
+
+    match item {
+        None => Err(LoadError::NoData),
+        Some(d) => {
+            let decoded = serde_json::from_str(&d);
+            decoded.map_err(LoadError::DecodeError)
+        }
+    }
+}
+
+/// Removes key and associated data from local storage
+pub fn remove_item(storage: &Storage, key: &str) {
+    let _ = storage.remove_item(key);
 }
 
-/// Load a store, to a deserializable data structure.
-pub fn load_data<T>(storage: &Storage, name: &str) -> Option<T>
+/// Things that can go wrong when trying to save data to local storage.
+pub enum SaveError {
+    /// The browser denied saving to local storage. Usually because the storage is full.
+    /// See: https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem#Exceptions
+    CouldNotSave,
```

Comment:
`SaveError` / `SetError`? Also we can save also exception content - `SetError(JsValue)` (applicable to all errors).